rfcs/45-stageless.md

```diff
@@ -0,0 +1,834 @@
+# Feature Name: `stageless`
+
+## Summary
+
+The existing stage boundaries result in a large number of complications: preventing us from operating across stages.
+Moreover, several related features (run criteria, states and fixed timesteps) are rather fragile and overly complex.
+This tangled mess needs to be holistically refactored: simplifying the scheduling model down into a single `Schedule`, moving towards an intent-based configuration style, taking full advantage of labels and leveraging exclusive systems to handle complex logic.

```

Comment:
"single `Schedule`" contradicts "each `App` can store multiple dynamically modifiable schedules" from the user-facing explanation.

Not sure what "leveraging exclusive systems to handle complex logic" means.